oval_graph/parts/footer.html

```diff
@@ -1,13 +1,137 @@
 </script>
+
 <body>
-    <div id="data"></div>
+	<div id="data"></div>
+	<div id="modal" class="modal">
+		<div class="modal-content">
+			<span id="close" class="close">&times;</span>
+			<div id="content"></div>
+		</div>
+	</div>
 	<script>
-	var data = JSON.parse(JSON.stringify(data_of_tree));
-	$('#data').jstree({
-            'core': {
-                'data': [data]
-            }
-		});	
+		function prepare_file_object(table_object) {
+			var permission = {
+				'uread': null,
+				'uwrite': null,
+				'uexec': null,
+				'gread': null,
+				'gwrite': null,
+				'gexec': null,
+				'oread': null,
+				'owrite': null,
+				'oexec': null
+			}
+			var new_table_object = {};
+			Object.keys(table_object).forEach(key => {
+				if (key in permission) {
+					permission[key] = table_object[key];
+				} else {
+					new_table_object[key] = table_object[key];
+				}
+			});
+			var out = '<code>';
+			Object.keys(permission).forEach(key => {
+				if (permission[key] == 'true') {
+					switch (key.substring(1, key.length)) {
+						case 'read':
+							out += "r";
+							break;
+						case 'write':
+							out += "w";
+							break;
+						case 'exec':
+							out += "x";
+							break;
+						default:
+							// pass
+							break;
+					}
+				} else {
+					out += "-";
+				}
+			});
+			out += '</code>';
+			new_table_object['permission'] = out;
+			return new_table_object;
+		}
+
+		function format_header(str) {
+			var index_special_char = str.indexOf('@');
+			var text = str.substring(0, index_special_char != -1 ? index_special_char : str.length);
+			var text_with_spaces = text.replace("_", " ");
+			return text_with_spaces.charAt(0).toUpperCase() + text_with_spaces.slice(1);
+		}
+
+		function create_table(type, table_object) {
+			switch (type) {
+				case 'file_object':
+					table_object = prepare_file_object(table_object);
```

Comment:
Try to check out if you can detect whether a function `prepare_<type>` exists, and if yes, apply it on the table_object. That way, you will get rid of the `switch` and you make addition of "js visualization plugins" easier.
If it's too complicated, let's do it in another PR.